theatre/studio/src/utils/invariant.ts

```diff
@@ -0,0 +1,93 @@
+import {devStringify} from './devStringify'
+
+type accepts = string | number | object
+
+/**
+ *
+ */
+export function invariant(
```

Comment:
That screws up the type system. We can instead just move that check to within the function, here. WDYT?

Reply:
I mean to say that invariant `asserts` the value in TypeScript's type space, so calls after this can actually expect that invariant condition is guaranteed. This benefit is removed if we have to wrap it in a separate node env check.

Additionally, adding env guards can make the code less readable.

I'd like to see us using invariant to flag for clearly invalid and irrecoverable cases such as a value missing from a map or other expectations that we have in the moment we're writing the code. This way, we can more confidently write convoluted control flow while getting context aware invariants.

My mentality is derived from [Matt Klein's article written while he worked on Envoy proxy](https://medium.com/@mattklein123/crash-early-and-crash-often-for-more-reliable-software-597738dd21c5) and [the Erlang approach](https://ferd.ca/the-zen-of-erlang.html) at large.